internal/toolexec/proxy/proxy.go

```diff
@@ -0,0 +1,120 @@
+package proxy
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+type (
+	// CommandType represents a Go compilation command type
+	// Currently support types are:
+	// - compile
+	// - link
+	// - unknown
```

Comment:
The "currently supported types" is more of an example IMHO... the actual list of supported stuff is best documented via the constants below.

```suggestion
	// CommandType represents a Go toolchain command type, such
	// as "compile", "link", etc...
```